### PR TITLE
lsp-save-logs: New function (Fixes #644).

### LINKED ISSUE #644: Supporting logging to files
The official LSP website comes with an [Inspector app ](https://microsoft.github.io/language-server-protocol/inspector/)that allows you to examine a detailed log of all requests/responses/notifications between a language server and a client. `lsp-mode` should support logging communication to a JSON file that can be read by the app, which would make debugging easier than it is now, for both developers and users.

### PR DESCRIPTION
**Pull request by vibhavp**

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -1416,9 +1423,72 @@ If WORKSPACE is not provided current workspace will be used."
...
+   :process-time process-time
+   :method method
+   :id id
+   :type type
+   :body (or arg (plist-get body :params))))
+
+(defun lsp-save-logs (file)
+  (interactive "F")
+  (let ((trace (reverse (lsp--client-message-trace (lsp--workspace-client
+                                                    (nth 0 (lsp-workspaces))))))
```
Comment (yyoncho) [inline lsp-mode.el]: we may interactively ask which workspace to save if there are more than one workspaces active

Comment (vibhavp) [reply lsp-mode.el]: Is there a way to map workspaces to a user-selectable string?

Comment (yyoncho) [reply lsp-mode.el]: yeah - lsp--workspace-print

Comment (vibhavp) [reply lsp-mode.el]: Done, thanks.

**Pull request merged**